src/controllers/asteroidsController.js

```
@@ -0,0 +1,13 @@
+import getAsteroidsData from "../services/asteroidsService.js"
+import getFormattedDate from "../utils/dateTimeUtils.js";
+
+export const getAsteroids = async (req, res) => {
+    try {
+        const {startDate, endDate} = getFormattedDate();
+        const data = await getAsteroidsData(startDate, endDate);
+        res.json(data);
+    } catch (error) {
+        console.error('Error fetching data:', error);
```

it would be nice to have an error handler to process the error in the separate module
Probably not a good example but you can see the idea:
```
export const errorHandler= (error, request, response) => {
  const status = error.status ?? 500;
  const message = error.message ?? 'Error fetching data from NASA API';

  response.status(status).json({ message: `${message}` });
};
```
Then you can use it in server.js:
```
import router from "./routers/asteroids.js";
**import { errorHandler } from './errors/error.handler.js';**

const PORT = process.env.PORT;
const app = express();
app.use(express.json());

app.use('/', router);

**app.use(errorHandler)**
```

And finally, you can propagate the error to next function:
```
export const getAsteroids = async (req, res, **next**) => {
    try {
        const {startDate, endDate} = getFormattedDate();
        const data = await getAsteroidsData(startDate, endDate);
        res.json(data);
    } catch (error) {
        **next(error)**
      }
}
```